src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -952,7 +1121,91 @@ internal static object ComObjectForInterface(IntPtr externalComObject)
                 throw new NotSupportedException(SR.InvalidOperation_ComInteropRequireComWrapperInstance);
             }
 
-            return s_globalInstanceForMarshalling.GetOrCreateObjectForComInstance(externalComObject, CreateObjectFlags.Unwrap);
+            // TrackerObject support and unwrapping matches the built-in semantics that the global marshalling scenario mimics.
+            return s_globalInstanceForMarshalling.GetOrCreateObjectForComInstance(externalComObject, CreateObjectFlags.TrackerObject | CreateObjectFlags.Unwrap);
+        }
+
+        internal static IntPtr GetOrCreateTrackerTarget(IntPtr externalComObject)
+        {
+            if (s_globalInstanceForTrackerSupport == null)
+            {
+                throw new NotSupportedException(SR.InvalidOperation_ComInteropRequireComWrapperInstance);
```

Comment:
Should all these be `ThrowHelper.ThrowNotSupportedException`? There's already an overload taking a resource.

Reply:
It looks like it needs an ExceptionResource rather than a string and from looking at the ones there it seems to be a limited set.  So leaving this alone for now based on the feedback in the other comment about it being meant for methods which are small hot methods.